flymake-vale.el

```diff
@@ -150,24 +173,24 @@ Passing the results and source BUF to CALLBACK."
   (when (process-live-p flymake-vale--proc)
     (kill-process flymake-vale--proc))
   (let* ((process-connection-type nil)
+         (callback flymake-vale--report-fnc)
+         (buf (current-buffer))
          (proc (apply #'start-process
                       "flymake-vale-process"
                       flymake-vale-output-buffer
                       flymake-vale-program
                       "--output"
                       "JSON"
-                      flymake-vale-program-args)))
+                      (append flymake-vale-program-args (flymake-vale--detect-extension (current-buffer))))))
```

Comment:
to be safe, this should probably be called in `flymake-vale--detect-extension` along with `flymake-vale--source-buffer`

Reply:
by "this" i mean the `(current-buffer)` expression